src/libs/TransactionUtils.js

```diff
@@ -79,7 +79,7 @@ function buildOptimisticTransaction(
  * @returns {Boolean}
  */
 function hasReceipt(transaction) {
-    return !_.isEmpty(lodashGet(transaction, 'receipt'));
+    return Boolean(lodashGet(transaction, 'receipt.source'));
```

Comment:
I think this should work better as the receiptID will be an empty string when this is incorrectly returned
```suggestion
    return !_.isEmpty(lodashGet(transaction, 'receipt')) && !_.isEmpty(lodashGet(transaction, 'receipt.receiptID'));
```

Reply:
yup, it does 😄

Reply:
Yo, sorry, lets just use the state then, that's set and once Auth is deployed tomorrow morning this wont be such a problem

We could also optimistically set receipt ID which will be overwritten with BE or is it used anywhere now?

Reply:
Updated